src/session.ts

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import utils = require('./utils')
+import kernel = require('./kernel')
+
+import Signal = phosphor.core.Signal;
+import emit = phosphor.core.emit;
+import IAjaxSuccess = utils.IAjaxSuccess;
+import IAjaxError = utils.IAjaxError;
+
+
+/**
+ * Notebook Identification specification.
+ */
+export
+interface INotebookId {
+  path: string;
+};
+
+
+/**
+ * Session Identification specification.
+ */
+export
+interface ISessionId {
+  id: string;
+  notebook: INotebookId;
+  kernel: kernel.IKernelId;
+};
+
+
+/**
+ * Session initialization options.
+ */
+export
+interface ISessionOptions {
+  notebookPath: string;
+  kernelName: string;
+  baseUrl: string;
+  wsUrl: string;
+};
+
+
+/**
+ * Session object for accessing the session REST api. The session
+ * should be used to start kernels and then shut them down -- for
+ * all other operations, the kernel object should be used.
+ **/
+export 
+class Session {
+
+  static statusChanged = new Signal<Session, string>();
+
+  /**
+   * GET /api/sessions
+   *
+   * Get a list of the current sessions.
+   */
+  static list(sessionServiceUrl: string): Promise<ISessionId[]> {
+    return utils.ajaxRequest(sessionServiceUrl, {
+      method: "GET",
+      dataType: "json"
+    }).then((success: IAjaxSuccess): ISessionId[] => {
+      if (success.xhr.status === 200) {
+        if (!Array.isArray(success.data)) {
+          throw Error('Invalid Session list');
+        }
+        for (var i = 0; i < success.data.length; i++) {
+          validateSessionId(success.data[i]);
+        }
+        return success.data;
+      }
+      throw Error('Invalid Status: ' + success.xhr.status);
+    });
+  }
+
+  /**
+   * Construct a new session.
+   */
+  constructor(options: ISessionOptions) {
+    this._notebookModel = {
+      path: options.notebookPath
+    };
+    this._kernelModel = {
+      id: null,
+      name: options.kernelName
+    };
+
+    this._baseUrl = options.baseUrl;
+    this._wsUrl = options.wsUrl;
+    this._sessionServiceUrl = utils.urlJoinEncode(this._baseUrl, 
+                                                  'api/sessions');
```

Comment:
Having the 'api/sessions' as a variable at the top of the file also makes the `list` method above easier - all you pass is the base url, and it can know to append `'api/sessions'`.